OpenSim/Common/Component.cpp

```diff
@@ -565,6 +565,9 @@ unsigned Component::printComponentsMatching(const std::string& substring) const
 
 int Component::getNumStateVariables() const
 {
+    // Must have already called initSystem.
+    OPENSIM_THROW_IF(!hasSystem(), ComponentHasNoSystem, getName().c_str());
```

Comment:
Consider using `OPENSIM_THROW_IF_FRMOBJ` so you don't to pass in the name and also report the concrete Class Name, which can be helpful.